---
title: "R Notebook"
editor_options: 
  chunk_output_type: console
---

```{r Load libraries, message=F}
if(!require(here, quietly = F)) {
  install.packages("here") 
  library(here)
}
if(!require(tidyverse, quietly = F)) {
  install.packages("tidyverse")
  library(tidyverse)
}
if(!require(edgeR, quietly = T)) {
  BiocManager::install("edgeR")
  library(edgeR)
}
if(!require(RUVSeq, quietly = T)) {
  BiocManager::install("RUVSeq")
  library(RUVseq)
}
if(!require(mdp, quietly = T)) {
  BiocManager::install("mdp")
  library(mdp) 
}
if(!require(RColorBrewer, quietly = T)) {
  install.packages("RColorBrewer")
  library(RColorBrewer) 
}
if(!require(pheatmap, quietly = T)) {
  install.packages("pheatmap")
  library(pheattmap)
}
if(!require(ggfortify, quietly = T)) {
  install.packages("ggfortify")
  library(ggfortify)
}
```

```{r Input files}
#Raw counts file
counts <- read.delim(here("data",
                          "gene_counts_merged_datasets_children_adults_elderly.txt"))

#Raw counts corrected with Combat-seq
#counts <- read.delim(here("results","gene_counts_batch-effect_correction_Adults-Children_UK_Gambia_UK-ICL_AgesBatch1_Combat-seq_with_nameofG.txt"))

#Raw counts corrected with Combat-seq curettage
#counts <- read.delim(here("data","gene_counts_merged_datasets_children_adults_elderly_curettage_Combat-seq.tsv"))

phenodata <- read.delim(here("data", "sample_information.txt"),
                        stringsAsFactors = T)

phenodata_curettage <- read.delim(here("data",
                                       "phenodata_curettage_102samples.tsv"), 
                                  stringsAsFactors = T)

tmm <- read.delim(here("data", "tmm_counts_curettage.tsv"))

tpm <- read.delim(here("results", 
                       "TPM_merged_datasets_children_adults_elderly.txt"))

degs <- read.delim(here("data","diff_expression_byAge_bySex_curettage.tsv"))
```


```{r MDP outliers}
phenodata <- phenodata[phenodata$timepoint == "baseline" & 
                       phenodata$sampling.method == "curettage",]

phenodata$Group <- factor(phenodata$Group, c("Children","Adults","Elderly"))

hgnc <- counts[,1:2]

rownames(counts) <- counts$ID

# MDP analysis
subcounts <- subset(counts, select = as.vector(phenodata$label))
keep <- rowSums(subcounts > 5) > ncol(subcounts)*0.3
subcounts <- subcounts[keep,]

mdp.results <- mdp(data = as.data.frame(cpm(subcounts)), 
                   pdata = data.frame(Sample = phenodata$label, 
                                      Class = phenodata$Group),
                   control_lab = "Children")

samplescore <- mdp.results$sample_scores$allgenes
 
samplescore %>% 
  ggplot(aes(x = Class, y = Score)) +
  geom_boxplot(coef=0.2407407)

rc <- boxplot.stats(samplescore[samplescore$Class == "Children",2])$out
ra <- boxplot.stats(samplescore[samplescore$Class == "Adults",2])$out
ro <- boxplot.stats(samplescore[samplescore$Class == "Elderly",2])$out

samples2drop.outliers <- subset(samplescore, Score %in% c(rc,ra,ro))

samples2drop.overlap <- subset(samplescore, 
                               Score > min(samplescore[samplescore$Class == "Elderly",2]) & 
                         Class != "Elderly")
samples2drop <- unique(c(samples2drop.outliers$Sample,
                         samples2drop.overlap$Sample))

# Remove samples based on MDP analysis
phenodata <- subset(phenodata, !(phenodata$label %in% samples2drop))
counts <- subset(counts, select = as.vector(phenodata$label))

# Number of samples by sex and age group
#phenodata %>% 
#  group_by(Group, sex) %>% 
#  summarise(n = n())

```

## Factors of unwanted variation were estimated for each comparison
```{r DEGs RUVseq Age + Sex with pair factors}
df <- NULL

for(i in levels(phenodata$Group)) {
  for(j in levels(phenodata$sex)) {
    
    subphenodata <- phenodata[phenodata$Group != i & phenodata$sex == j,]    
    subphenodata$Group <- factor(subphenodata$Group, 
                                 as.vector(unique(subphenodata$Group)))
    
    subcounts <- subset(counts, select = as.vector(subphenodata$label))
    
    #Filter low expression genes
    keep <- rowSums(subcounts > 5) > ncol(subcounts)*0.3
    subcounts <- subcounts[keep,]
    
    subhgnc <- hgnc[hgnc$ID %in% rownames(subcounts),]
    
    ncounts <- newSeqExpressionSet(as.matrix(subcounts),
                                   phenoData = data.frame(Group = subphenodata$Group,
                                                          row.names=subphenodata$label))
    
    ### Empirical stable genes
    design <- model.matrix(~Group, data=pData(ncounts))
    y <- DGEList(counts=counts(ncounts), group=subphenodata$Group)
    y <- calcNormFactors(y, method="upperquartile")
    y <- estimateGLMCommonDisp(y, design)
    y <- estimateGLMTagwiseDisp(y, design)
    fit <- glmFit(y, design)
    lrt <- glmLRT(fit, coef=2)
    top <- topTags(lrt, n=nrow(ncounts))$table
    empirical <- rownames(ncounts)[which(!(rownames(ncounts) %in% 
                                             rownames(top)[1:5000]))]
    ###
    ncounts <- betweenLaneNormalization(ncounts, which="upper")
    
    set1 <- RUVg(ncounts, empirical, k=1)
    
    design <- model.matrix(~Group + W_1, data=pData(set1))
    
    y <- DGEList(counts = subcounts, genes = subhgnc, group = subphenodata$Group)
    y <- calcNormFactors(y)
    y <- estimateGLMCommonDisp(y, design)
    y <- estimateGLMTagwiseDisp(y, design)
    fit <- glmFit(y, design, robust=T)
    degs <- topTags(glmLRT(fit, coef=2), n=NULL)$table
    
    ndown <- nrow(degs[degs$logFC < -1 & degs$FDR <0.01,])
    nup <- nrow(degs[degs$logFC > 1 & degs$FDR <0.01,])
    
    filename <- paste0("DEGs_", levels(subphenodata$Group)[1], ".vs.", 
                       levels(subphenodata$Group)[2],
                       ".sex_",j,"_glmLRT_RUVg_empirical_control.tsv")
    
    write.table(degs, here("results", "diff_express_edgeR" ,filename),
                sep='\t',quote=F,row.names = F)
    
    df <- rbind(df, data.frame(ref = levels(subphenodata$Group)[1], 
                               tret= levels(subphenodata$Group)[2],
                               sex = j, Up = nup, Down = ndown))
  }
}

write.table(df, here("results", "diff_express_edgeR",
                     "number_DEGS_logFC1_fdr0.01_glmLRT_RUVg_empirical_control.txt"),
                sep='\t',quote=F,row.names = F)
``` 

The first round of diff express analysis was done with 113 samples. After
normalization with RUVSeq some outliers were detected on PCA plot. To identify and remove these outliers I applied the functions in the chunk below. It was necessary two rounds of sample removing in order to have a PCA representing a homogeneous set of samples. The final list of samples was saved at 
"phenodata_curettage_102samples.tsv".

```{r Outlier detection}
#from https://www.r-bloggers.com/detecting-outlier-samples-in-pca/
U <- prcomp.ruv.norm$x

ind.out <- apply(U, 2, 
                 function(x) which( (abs(x - median(x)) / mad(x)) > 6 )) %>%
  Reduce(union, .) %>%
  print()

col <- rep("black", nrow(U)); col[ind.out] <- "red"
qplot(U[, 1], U[, 2], color = I(col), size = I(2)) + coord_equal()

#phenodata_curettage <- phenodata_curettage[-ind.out,]
#phenodata_curettage <- phenodata_curettage[-ind.out2,]

#write.table(phenodata_curettage, 
#            here("data","phenodata_curettage_102samples.tsv"),
#            sep = "\t", quote=F, row.names = F)
```

## Factors of unwanted variation were estimated for all samples at once
```{r DEG RUVseq Age + Sex with global factors}
colors <- brewer.pal(3, "Set2")

hgnc <- counts[,1:2]

rownames(counts) <- counts$ID

subcounts <- subset(counts, select = as.vector(phenodata_curettage$label))

#Filter low expression genes
keep <- rowSums(subcounts > 1) > ncol(subcounts)*0.5
subcounts <- subcounts[keep,]

subhgnc <- hgnc[hgnc$ID %in% rownames(subcounts),]

ncounts <- newSeqExpressionSet(as.matrix(subcounts),
                               phenoData = data.frame(Group = phenodata_curettage$Group,
                                                      sex = phenodata_curettage$sex,
                                                      row.names=phenodata_curettage$label))

### Empirical control genes
design <- model.matrix(~Group, data=pData(ncounts))
y <- DGEList(counts=counts(ncounts), group=phenodata_curettage$Group)
y <- calcNormFactors(y, method="upperquartile")
y <- estimateGLMCommonDisp(y, design)
y <- estimateGLMTagwiseDisp(y, design)
fit <- glmFit(y, design)
lrt <- glmLRT(fit, coef=2:3)
top <- topTags(lrt, n=nrow(ncounts))$table
empirical <- rownames(ncounts)[which(!(rownames(ncounts) %in% 
                                         rownames(top)[1:10000]))]
###
ncounts <- betweenLaneNormalization(ncounts, which="upper")

### RUVseq norm
ruv.norm <- RUVg(ncounts, empirical, k=4)

#ruvcounts <- normCounts(ruv.norm)
#ruvcounts <- merge(subhgnc, ruvcounts, by.x = "ID", by.y = "row.names")
#write.table(ruvcounts, here("data","ruv_counts_curettage.tsv"),
#            sep = "\t", quote=F, row.names = F)

#Plot of Relative Log Expression
plotRLE(ruv.norm, col = colors[factor(phenodata_curettage$Group, 
                                    levels= c("Children","Adults", "Elderly"))], 
        outline=FALSE, ylim=c(-4, 4), main = "RUV norm",
        ylab= "Relative Log Expression")

#plotPCA use top 500 most variant genes before prcomp, using prcomp directly instead
#plotPCA(ruv.norm, 
#        col=colors[factor(phenodata_curettage$Group, 
#                          levels= c("Children","Adults", "Elderly"))],
#        cex=1.2)

# PCA plot
ruv.norm.t <- as.data.frame(t(normCounts(ruv.norm)))
ruv.norm.t$label <- rownames(ruv.norm.t)
ruv.norm.t <- merge(phenodata_curettage, ruv.norm.t, by="label")

prcomp.ruv.norm <- prcomp(ruv.norm.t[,-c(1:7)])

autoplot(prcomp.ruv.norm , x= 1,y = 2,ruv.norm.t[,1:7], shape = 'sex', 
         colour= 'Group', size=3) +
  scale_color_manual(values= brewer.pal(3,"Set2")) +
  labs(title = "RUV norm / k = 4 / 102 samples") +
  theme_bw(base_size = 15,)


df <- NULL

for(i in levels(phenodata_curettage$Group)) {
  for(j in levels(phenodata_curettage$sex)) {
    
    subphenodata <- phenodata_curettage[phenodata_curettage$Group != i &
                                     phenodata_curettage$sex == j,]
    subphenodata$Group <- factor(subphenodata$Group, 
                                 as.vector(unique(subphenodata$Group)))
    
    samples <- subphenodata[,1]
    
    dpheno <- pData(ruv.norm)[samples,]
    dpheno$Group <- factor(dpheno$Group, unique(dpheno$Group))
    
    design <- model.matrix(~Group + W_1 + W_2 + W_3 + W_4, data = dpheno)
    
    y <- DGEList(counts = counts(ruv.norm)[,samples], genes = subhgnc)
    y <- calcNormFactors(y)
    y <- estimateGLMCommonDisp(y, design)
    y <- estimateGLMTagwiseDisp(y, design)
    fit <- glmFit(y, design, robust=T)
    degs <- topTags(glmLRT(fit, coef=2), n=NULL)$table
    
    ndown <- nrow(degs[degs$logFC < -1 & degs$FDR <0.01,])
    nup <- nrow(degs[degs$logFC > 1 & degs$FDR <0.01,])
        
    filename <- paste0("DEGs_", levels(subphenodata$Group)[1], ".vs.",
                       levels(subphenodata$Group)[2],
                       ".sex_",j,"_glmLRT_RUVg_k4_empirical_control_all_groups_102samples.tsv")
        
    write.table(degs, here("results", "diff_express_edgeR" ,filename),
                sep='\t',quote=F,row.names = F)
        
    df <- rbind(df, data.frame(ref = levels(subphenodata$Group)[1], 
                               tret= levels(subphenodata$Group)[2],
                               sex = j, Up = nup, Down = ndown))
  }
}

write.table(df, here("results", "diff_express_edgeR",
                     "number_DEGS_logFC1_fdr0.01_glmLRT_RUVg_k4_empirical_control_all_groups_102samples.txt"),
                sep='\t',quote=F,row.names = F)

#Plot expression
ruv.norm.t <- as.data.frame(t(normCounts(ruv.norm)))

ruv.norm.t$label <- rownames(ruv.norm.t)
ruv.norm.t <- merge(phenodata_curettage, ruv.norm.t, by="label")
x <- ruv.norm.t %>% 
  pivot_longer(-colnames(ruv.norm.t[,1:7]), 
               names_to = "ensemblID", values_to = "values") %>%
  left_join(hgnc, by= c("ensemblID" = "ID"))

genes <- c("ACE2","TMPRSS2","FURIN","ANPEP","CTSL","CTSB")
genes <- c("TMPRSS2","FURIN")

x %>%
  mutate(values = log2(values + 1)) %>% 
  filter(nameofG %in% genes) %>%
  mutate(Group = fct_relevel(Group, "Children","Adults","Elderly")) %>%
  mutate(nameofG = fct_relevel(nameofG, genes)) %>%
  ggplot(aes(x = Group, y = values, fill = sex), cex=1.5) +
  geom_boxplot(position= "dodge", outlier.shape = NA) +
  facet_wrap(vars(nameofG),ncol =3) +
  ylab("log2 RUV norm")+
  labs(title = "k = 4 / 102 samples")+
  theme_bw(base_size = 12)
  scale_y_continuous(limits = c(9,12))

####### Calculate expression by SEX within groups

df <- NULL

for(i in levels(phenodata_curettage$Group)) {
    
    subphenodata <- phenodata_curettage[phenodata_curettage$Group == i,]

    samples <- subphenodata[,1]
    
    dpheno <- pData(ruv.norm)[samples,]
    dpheno$Group <- factor(dpheno$Group, unique(dpheno$Group))
    
    design <- model.matrix(~sex + W_1 + W_2 + W_3 + W_4, data = dpheno)
    
    y <- DGEList(counts = counts(ruv.norm)[,samples], genes = subhgnc)
    y <- calcNormFactors(y)
    y <- estimateGLMCommonDisp(y, design)
    y <- estimateGLMTagwiseDisp(y, design)
    fit <- glmFit(y, design, robust=T)
    degs <- topTags(glmLRT(fit, coef=2), n=NULL)$table
    
    ndown <- nrow(degs[degs$logFC < -1 & degs$FDR <0.01,])
    nup <- nrow(degs[degs$logFC > 1 & degs$FDR <0.01,])
        
    filename <- paste0("DEGs_", levels(subphenodata$sex)[1], ".vs.",
                       levels(subphenodata$sex)[2], "_", levels(dpheno$Group), "_glmLRT_RUVg_k4_empirical_control_all_groups_102samples.tsv")
        
    write.table(degs, here("results", "diff_express_edgeR" ,filename),
                sep='\t',quote=F,row.names = F)
        
    df <- rbind(df, data.frame(ref = levels(subphenodata$sex)[1], 
                               tret= levels(subphenodata$sex)[2],
                               Group = levels(dpheno$Group), 
                               Up = nup, Down = ndown))
}

write.table(df, here("results", "diff_express_edgeR",
                     "number_DEGS_logFC1_fdr0.01_glmLRT_RUVg_k4_empirical_control_all_groups_102samples_FvsM.txt"),
                sep='\t',quote=F,row.names = F)  
  
out <- as.data.frame(normCounts(ruv.norm))

out$ID <- rownames(out)
out <- merge(subhgnc, out, by="ID")

write.table(out, here("results","counts_ruv-norm_k3.tsv"), 
            sep = "\t", quote = F, row.names = F)
``` 

## Factors of unwanted variation estimated for all samples at once and comparison just between age groups
```{r DEG RUVseq Age with global factors}
colors <- brewer.pal(3, "Set2")

hgnc <- counts[,1:2]

rownames(counts) <- counts$ID

subcounts <- subset(counts, select = as.vector(phenodata_curettage$label))

#Filter low expression genes
keep <- rowSums(subcounts > 1) > ncol(subcounts)*0.5
subcounts <- subcounts[keep,]

subhgnc <- hgnc[hgnc$ID %in% rownames(subcounts),]

ncounts <- newSeqExpressionSet(as.matrix(subcounts),
                               phenoData = data.frame(Group = phenodata_curettage$Group,
                                                      sex = phenodata_curettage$sex,
                                                      row.names=phenodata_curettage$label))

### Empirical control genes
design <- model.matrix(~Group, data=pData(ncounts))
y <- DGEList(counts=counts(ncounts), group=phenodata_curettage$Group)
y <- calcNormFactors(y, method="upperquartile")
y <- estimateGLMCommonDisp(y, design)
y <- estimateGLMTagwiseDisp(y, design)
fit <- glmFit(y, design)
lrt <- glmLRT(fit, coef=2:3)
top <- topTags(lrt, n=nrow(ncounts))$table
empirical <- rownames(ncounts)[which(!(rownames(ncounts) %in% 
                                         rownames(top)[1:10000]))]
###
ncounts <- betweenLaneNormalization(ncounts, which="upper")

### RUVseq norm
ruv.norm <- RUVg(ncounts, empirical, k=4)

#ruvcounts <- normCounts(ruv.norm)
#ruvcounts <- merge(subhgnc, ruvcounts, by.x = "ID", by.y = "row.names")
#write.table(ruvcounts, here("data","ruv_counts_curettage.tsv"),
#            sep = "\t", quote=F, row.names = F)

#Plot of Relative Log Expression
plotRLE(ruv.norm, col = colors[factor(phenodata_curettage$Group, 
                                    levels= c("Children","Adults", "Elderly"))], 
        outline=FALSE, ylim=c(-4, 4), main = "RUV norm",
        ylab= "Relative Log Expression")

#plotPCA use top 500 most variant genes before prcomp, using prcomp directly instead
#plotPCA(ruv.norm, 
#        col=colors[factor(phenodata_curettage$Group, 
#                          levels= c("Children","Adults", "Elderly"))],
#        cex=1.2)

# PCA plot
ruv.norm.t <- as.data.frame(t(normCounts(ruv.norm)))
ruv.norm.t$label <- rownames(ruv.norm.t)
ruv.norm.t <- merge(phenodata_curettage, ruv.norm.t, by="label")

prcomp.ruv.norm <- prcomp(ruv.norm.t[,-c(1:7)])

autoplot(prcomp.ruv.norm , x= 1,y = 2,ruv.norm.t[,1:7], shape = 'sex', 
         colour= 'Group', size=3) +
  scale_color_manual(values= brewer.pal(3,"Set2")) +
  labs(title = "RUV norm / k = 4 / 102 samples") +
  theme_bw(base_size = 15,)


df <- NULL

for(i in levels(phenodata_curettage$Group)) {
    
    subphenodata <- phenodata_curettage[phenodata_curettage$Group != i,]
    subphenodata$Group <- factor(subphenodata$Group, 
                                 as.vector(unique(subphenodata$Group)))
    
    samples <- subphenodata[,1]
    
    dpheno <- pData(ruv.norm)[samples,]
    dpheno$Group <- factor(dpheno$Group, unique(dpheno$Group))
    
    design <- model.matrix(~Group + W_1 + W_2 + W_3 + W_4, data = dpheno)
    
    y <- DGEList(counts = counts(ruv.norm)[,samples], genes = subhgnc)
    y <- calcNormFactors(y)
    y <- estimateGLMCommonDisp(y, design)
    y <- estimateGLMTagwiseDisp(y, design)
    fit <- glmFit(y, design, robust=T)
    degs <- topTags(glmLRT(fit, coef=2), n=NULL)$table
    
    ndown <- nrow(degs[degs$logFC < -1 & degs$FDR <0.01,])
    nup <- nrow(degs[degs$logFC > 1 & degs$FDR <0.01,])
        
    filename <- paste0("DEGs_", levels(subphenodata$Group)[1], ".vs.",
                       levels(subphenodata$Group)[2],
              "_glmLRT_RUVg_k4_empirical_control_all_groups_102samples.tsv")
        
    write.table(degs, here("results", "diff_express_edgeR" ,filename),
                sep='\t',quote=F,row.names = F)
        
    df <- rbind(df, data.frame(ref = levels(subphenodata$Group)[1], 
                               treat= levels(subphenodata$Group)[2],
                              Up = nup, Down = ndown))
}

write.table(df, here("results", "diff_express_edgeR",
                     "number_DEGS_logFC1_fdr0.01_glmLRT_RUVg_k4_empirical_control_all_groups_102samples_only_byAge.txt"),
                sep='\t',quote=F,row.names = F)
```


```{r DEGs standard edgeR}
design <- model.matrix(~0 + Group:sex, data=phenodata)
    
y <- DGEList(counts = counts, genes = hgnc,
             group = interaction(phenodata$Group, phenodata$sex))

keep <- filterByExpr(y)
y <- y[keep, , keep.lib.sizes=FALSE]

y <- calcNormFactors(y,method=c("TMM"))
y <- estimateGLMCommonDisp(y, design)
y <- estimateGLMTagwiseDisp(y, design)
fit <- glmQLFit(y, design, robust=TRUE)

#F
adults.vs.children.f <- topTags(glmQLFTest(fit, contrast=c(-1,1,0,0,0,0)),
                                n=NULL)$table

elderly.vs.adults.f <- topTags(glmQLFTest(fit, contrast=c(0,-1,1,0,0,0)),
                               n=NULL)$table

elderly.vs.children.f <- topTags(glmQLFTest(fit, contrast=c(-1,0,1,0,0,0)),
                                 n=NULL)$table

#M
adults.vs.children.m <- topTags(glmQLFTest(fit, contrast=c(0,0,0,-1,1,0)),
                                n=NULL)$table

elderly.vs.adults.m <- topTags(glmQLFTest(fit, contrast=c(0,0,0,0,-1,1)),
                               n=NULL)$table

elderly.vs.children.m <- topTags(glmQLFTest(fit, contrast=c(0,0,0,-1,0,1)),
                                 n=NULL)$table

#M.vs.F
m.vs.f.children <- topTags(glmQLFTest(fit, contrast=c(-1,0,0,1,0,0)),
                                 n=NULL)$table
m.vs.f.adults <- topTags(glmQLFTest(fit, contrast=c(0,-1,0,0,1,0)),
                                 n=NULL)$table
m.vs.f.elderly <- topTags(glmQLFTest(fit, contrast=c(0,0,-1,0,0,1)),
                                 n=NULL)$table

out <- adults.vs.children.f %>%
  full_join(adults.vs.children.m, by = c("ID","nameofG"), 
            suffix = c(".Adults.vs.Children.F", ".Adults.vs.Children.M")) %>%
  full_join(elderly.vs.children.f, by = c("ID","nameofG")) %>%
  full_join(elderly.vs.children.m, by = c("ID","nameofG"),
            suffix = c(".Elderly.vs.Children.F", ".Elderly.vs.Children.M")) %>%
  full_join(elderly.vs.adults.f, by = c("ID","nameofG")) %>%
  full_join(elderly.vs.adults.m, by = c("ID","nameofG"),
            suffix = c(".Elderly.vs.Adults.F", ".Elderly.vs.Adults.M")) %>%
  full_join(m.vs.f.children, by = c("ID","nameofG")) %>%
  full_join(m.vs.f.adults, by = c("ID","nameofG"), 
            suffix = c(".M.vs.F.Children", ".M.vs.F.Adults")) %>%
  full_join(dplyr::rename(m.vs.f.elderly, logFC.M.vs.F.Elderly = logFC,
                          logCPM.M.vs.F.Elderly = logCPM, F.M.vs.F.Elderly = F,
                          PValue.M.vs.F.Elderly = PValue, FDR.M.vs.F.Elderly = FDR), 
            by = c("ID","nameofG"))


up.elderly.vs.children.f <- nrow(elderly.vs.children.f[elderly.vs.children.f$logFC > 1 &
                                                       elderly.vs.children.f$FDR < 0.01,])
down.elderly.vs.children.f <- nrow(elderly.vs.children.f[elderly.vs.children.f$logFC < -1 &
                                                         elderly.vs.children.f$FDR < 0.01,])
up.elderly.vs.children.m <- nrow(elderly.vs.children.m[elderly.vs.children.m$logFC > 1 &
                                                       elderly.vs.children.m$FDR < 0.01,])
down.elderly.vs.children.m <- nrow(elderly.vs.children.m[elderly.vs.children.m$logFC < -1 &
                                                         elderly.vs.children.m$FDR < 0.01,])

up.elderly.vs.adults.f <- nrow(elderly.vs.adults.f[elderly.vs.adults.f$logFC > 1 &
                                                         elderly.vs.adults.f$FDR < 0.01,])
down.elderly.vs.adults.f <- nrow(elderly.vs.adults.f[elderly.vs.adults.f$logFC < -1 &
                                                         elderly.vs.adults.f$FDR < 0.01,])
up.elderly.vs.adults.m <- nrow(elderly.vs.adults.m[elderly.vs.adults.m$logFC > 1 &
                                                         elderly.vs.adults.m$FDR < 0.01,])
down.elderly.vs.adults.m <- nrow(elderly.vs.adults.m[elderly.vs.adults.m$logFC < -1 &
                                                         elderly.vs.adults.m$FDR < 0.01,])

up.adults.vs.children.f <- nrow(adults.vs.children.f[adults.vs.children.f$logFC > 1 &
                                                         adults.vs.children.f$FDR < 0.01,])
down.adults.vs.children.f <- nrow(adults.vs.children.f[adults.vs.children.f$logFC < -1 &
                                                         adults.vs.children.f$FDR < 0.01,])
up.adults.vs.children.m <- nrow(adults.vs.children.m[adults.vs.children.m$logFC > 1 &
                                                         adults.vs.children.m$FDR < 0.01,])
down.adults.vs.children.m <- nrow(adults.vs.children.m[adults.vs.children.m$logFC < -1 &
                                                         adults.vs.children.m$FDR < 0.01,])

up.m.vs.f.children <- nrow(m.vs.f.children[m.vs.f.children$logFC > 1 &
                                             m.vs.f.children$FDR < 0.01,])
down.m.vs.f.children <- nrow(m.vs.f.children[m.vs.f.children$logFC < -1 &
                                             m.vs.f.children$FDR < 0.01,])
up.m.vs.f.adults <- nrow(m.vs.f.adults[m.vs.f.adults$logFC > 1 &
                                             m.vs.f.adults$FDR < 0.01,])
down.m.vs.f.adults <- nrow(m.vs.f.adults[m.vs.f.adults$logFC < -1 &
                                             m.vs.f.adults$FDR < 0.01,])
up.m.vs.f.elderly <- nrow(m.vs.f.elderly[m.vs.f.elderly$logFC > 1 &
                                             m.vs.f.elderly$FDR < 0.01,])
down.m.vs.f.elderly <- nrow(m.vs.f.elderly[m.vs.f.elderly$logFC < -1 &
                                             m.vs.f.elderly$FDR < 0.01,])


compar <- c(rep(c("Adults.vs.Children"),2), 
            rep(c("Elderly.vs.Children"),2), 
            rep(c("Elderly.vs.Adults"),2),
            rep(c("M.vs.F"),3))

up <- c(up.adults.vs.children.f, up.adults.vs.children.m,
        up.elderly.vs.children.f, up.elderly.vs.children.m,
        up.elderly.vs.adults.f, up.elderly.vs.adults.m,
        up.m.vs.f.children, up.m.vs.f.adults, up.m.vs.f.elderly)

down <- c(down.adults.vs.children.f, down.adults.vs.children.m,
          down.elderly.vs.children.f, down.elderly.vs.children.m,
          down.elderly.vs.adults.f, down.elderly.vs.adults.m,
        down.m.vs.f.children, down.m.vs.f.adults, down.m.vs.f.elderly)

df <- data.frame(Comparison = compar,
                 Sex = c(rep(c("F","M"),3), c("Children","Adults","Elderly")),
                 Up = up,
                 Down = down)

#number of diff express genes 
write.table(df, here("results","diff_express_edgeR",
                     "number_DEGS_logFC1_fdr0.01_standard.tsv"), 
            sep="\t",quote=F,row.names = F)

#write.table(adults.vs.children.f, here("results","diff_express_edgeR",
#                                       "DEGs_Children.vs.Adults.sex_F_standard.tsv"),
#            sep = "\t",quote=F,row.names=F)
#write.table(adults.vs.children.m, here("results","diff_express_edgeR",
#                                       "DEGs_Children.vs.Adults.sex_M_standard.tsv"),
#            sep = "\t",quote=F,row.names=F)
#write.table(elderly.vs.adults.m, here("results","diff_express_edgeR",
#                                       "DEGs_Adults.vs.Elderly.sex_M_standard.tsv"),
#            sep = "\t",quote=F,row.names=F)
#write.table(elderly.vs.adults.f, here("results","diff_express_edgeR",
#                                       "DEGs_Adults.vs.Elderly.sex_F_standard.tsv"),
#            sep = "\t",quote=F,row.names=F)
#write.table(elderly.vs.children.m, here("results","diff_express_edgeR",
#                                       "DEGs_Children.vs.Elderly.sex_M_standard.tsv"),
#            sep = "\t",quote=F,row.names=F)
#write.table(elderly.vs.children.f, here("results","diff_express_edgeR",
#                                       "DEGs_Children.vs.Elderly.sex_F_standard.tsv"),
#            sep = "\t",quote=F,row.names=F)
#write.table(m.vs.f.children, here("results","diff_express_edgeR",
#                                       "DEGs_M.vs.F.children_standard.tsv"),
#            sep = "\t",quote=F,row.names=F)
#write.table(m.vs.f.adults, here("results","diff_express_edgeR",
#                                       "DEGs_M.vs.F.adults_standard.tsv"),
#            sep = "\t",quote=F,row.names=F)
#write.table(m.vs.f.elderly, here("results","diff_express_edgeR",
#                                       "DEGs_M.vs.F.elderly_standard.tsv"),
#            sep = "\t",quote=F,row.names=F)

#write.table(out, here("results","diff_express_edgeR","diff_expression_byAge_bySex_curettage.tsv"),
#            sep='\t', quote=F, row.names=F)

#write.table(phenodata, here("data","phenodata_curettage.tsv"),sep='\t',quote=F,row.names=F)

#write.table(counts, here("data","raw_counts_curettage.tsv"),sep='\t',quote=F,row.names=F)
```


```{r Plots}
#Filter low expression genes
keep <- rowSums(counts > 5) > ncol(counts)*0.3
subcounts <- counts[keep,]

subhgnc <- hgnc[hgnc$ID %in% rownames(subcounts),]

ncounts <- newSeqExpressionSet(as.matrix(subcounts), 
                               phenoData = data.frame(Group = phenodata$Group,
                                                      row.names=phenodata$label))

### Empirical stable genes
design <- model.matrix(~Group, data=pData(ncounts))
y <- DGEList(counts=counts(ncounts), group=phenodata$Group)
y <- calcNormFactors(y, method="upperquartile")
y <- estimateGLMCommonDisp(y, design)
y <- estimateGLMTagwiseDisp(y, design)
fit <- glmFit(y, design)
lrt <- glmLRT(fit, coef=2:3)
top <- topTags(lrt, n=nrow(ncounts))$table
empirical <- rownames(ncounts)[which(!(rownames(ncounts) %in% rownames(top)[1:5000]))]
###

ncounts <- betweenLaneNormalization(ncounts, which="upper")

set1 <- RUVg(ncounts, empirical, k=1)

## Commands to save norm counts for all libraries

normcounts <- as.data.frame(t(cpm(normCounts(set1))))
normcounts$label <- rownames(normcounts)
normcounts <- merge(phenodata, normcounts, by="label")
x <- normcounts %>% 
  pivot_longer(-colnames(normcounts[,1:7]), 
               names_to = "ensemblID", values_to = "values") %>%
  left_join(hgnc, by= c("ensemblID" = "ID"))
#write.table(x, here("results",
#                    "RUVseq_adjusted_merged_datasets_children_adults_elderly_curettage.txt"), #
#            sep = "\t", quote = F, row.names=F)


#"RUVseq_adjusted_merged_datasets_children_adults_elderly_curettage.txt"))
genes <- c("ACE2","TMPRSS2","FURIN","ANPEP","DPP4","APOD","L3MBTL2","RBCK1","TMEM199","WDR55")

x %>%
  filter(nameofG %in% genes) %>%
  mutate(Group = fct_relevel(Group, "Children","Adults","Older")) %>%
  mutate(nameofG = fct_relevel(nameofG, genes)) %>%
  ggplot(aes(x = Group, y = values, fill = sex), cex=1.5) +
  geom_boxplot(position= "dodge", outlier.shape=NA) +
  facet_wrap(vars(nameofG),nrow=2) +
 # scale_y_continuous(trans='log2') +
  ylab("RUVSeq normalized counts")+
  theme_bw(base_size = 12) +
  ylim(0,350)


### TPM
rownames(tpm) <- tpm$ID
tpm <- subset(tpm, select= as.vector(phenodata$label))
tpm <- as.data.frame(t(tpm))

tpm$label <- rownames(tpm)
tpm <- merge(phenodata, tpm, by="label")
x <- tpm %>% 
  pivot_longer(-colnames(tpm[,1:7]), 
               names_to = "ensemblID", values_to = "values") %>%
  left_join(hgnc, by= c("ensemblID" = "ID"))


genes <- c("ACE2","TMPRSS2","FURIN","ANPEP","DPP4",
           "APOD","L3MBTL2","RBCK1","TMEM199","WDR55")

x %>%
  filter(nameofG %in% genes) %>%
  mutate(Group = fct_relevel(Group, "Children","Adults","Older")) %>%
  mutate(nameofG = fct_relevel(nameofG, genes)) %>%
  ggplot(aes(x = Group, y = values, fill = sex), cex=1.5) +
  geom_boxplot(position= "dodge", outlier.shape=NA) +
  facet_wrap(vars(nameofG),nrow=2) +
 # scale_y_continuous(trans='log2') +
  ylab("TPM")+
  theme_bw(base_size = 12) +
  ylim(0,350)

## TMM
countsdge <- DGEList(counts = counts, genes = hgnc, group = phenodata$Group)

countsdge <- calcNormFactors(countsdge, method = "TMM")
tmm <- as.data.frame(cpm(countsdge))
#tmm$ID <- rownames(tmm)
#tmm <- merge(hgnc, tmm, by="ID")
tmm <- as.data.frame(t(tmm))
#tmm <- as.data.frame(t(tmm[,-c(1:2)]))
#write.table(tmm, here("data", "tmm_counts_curettage"), 
#            sep="\t", quote=F, row.names=F)

tmm$label <- rownames(tmm)
tmm <- merge(phenodata, tmm, by="label")
x <- tmm %>% 
  pivot_longer(-colnames(tmm[,1:7]), 
               names_to = "ensemblID", values_to = "values") %>%
  left_join(hgnc, by= c("ensemblID" = "ID"))

#genes <- c("ACE2","TMPRSS2","FURIN","ANPEP","DPP4","CTSB","CTSL","APOD","L3MBTL2","RBCK1","TMEM199","WDR55")

genes <- c("ACE2","TMPRSS2","FURIN","ANPEP","CTSB","CTSL")

#negative controles
#genes <- c("ATP5G3","NDUFB11","UQCRFS1","COL3A1","COL1A1","COL4A5","TFRC","GOT1","NDUFB5","PTPRO","NDUFS3"	,"CRHBP","MYO5C","NDUFB6","GOT2","ATP5A1","ATP5G1","ATP5J","SDHB","SDHD")

x %>%
  filter(nameofG %in% genes) %>%
  mutate(Group = fct_relevel(Group, "Children","Adults","Elderly")) %>%
  mutate(nameofG = fct_relevel(nameofG, genes)) %>%
  ggplot(aes(x = Group, y = values, fill = sex), cex=1.5) +
  geom_boxplot(position= "dodge", outlier.shape=NA) +
  facet_wrap(vars(nameofG),nrow=2) +
  scale_y_continuous(trans='log2') +
  ylab("TMM normalized counts")+
  theme_bw(base_size = 12) 
  #ylim(0,350)

```

```{r TMM by group}
rownames(tmm) <- tmm$ID
tmm <- tmm[,-c(1:2)]
tmm <- as.data.frame(t(tmm))

tmm$label <- rownames(tmm)

out <- phenodata %>%
   full_join(tmm, by = "label") %>%
   pivot_longer(-colnames(df[,c(1:7)]), names_to = "genes", 
                values_to = "values") %>%
   group_by(Group, sex, genes) %>%
   summarise(mean=mean(values)) %>%
   pivot_wider(id_cols = "genes", names_from = c("Group","sex"), 
               values_from = "mean") %>%
   inner_join(degs, by = c("genes" = "ID")) %>%
   select(-starts_with(c("logCPM.", "F."))) %>%
   rename(ID = genes, TMM_Adults_M = Adults_M, TMM_Adults_F = Adults_F, 
          TMM_Children_F = Children_F, TMM_Children_M = Children_M, 
          TMM_Elderly_M = Older_M, TMM_Elderly_F = Older_F) %>%
   select(ID, nameofG, everything())

write.table(out, here("data","TMM_expression_byAge_bySex_curettage.tsv"), 
            sep = "\t", quote = F, row.names = F)
```

```{r Join RUVseq results}
adults.vs.children.f <- read.delim(here("results","diff_express_edgeR","DEGs_Children.vs.Adults.sex_F_glmLRT_RUVg_k4_empirical_control_all_groups_102samples.tsv"))
adults.vs.children.m <- read.delim(here("results","diff_express_edgeR","DEGs_Children.vs.Adults.sex_M_glmLRT_RUVg_k4_empirical_control_all_groups_102samples.tsv"))
elderly.vs.children.f <- read.delim(here("results","diff_express_edgeR","DEGs_Children.vs.Elderly.sex_F_glmLRT_RUVg_k4_empirical_control_all_groups_102samples.tsv"))
elderly.vs.children.m <- read.delim(here("results","diff_express_edgeR","DEGs_Children.vs.Elderly.sex_M_glmLRT_RUVg_k4_empirical_control_all_groups_102samples.tsv"))
elderly.vs.adults.f <- read.delim(here("results","diff_express_edgeR","DEGs_Adults.vs.Elderly.sex_F_glmLRT_RUVg_k4_empirical_control_all_groups_102samples.tsv"))
elderly.vs.adults.m <- read.delim(here("results","diff_express_edgeR","DEGs_Adults.vs.Elderly.sex_M_glmLRT_RUVg_k4_empirical_control_all_groups_102samples.tsv"))


out2 <- adults.vs.children.f %>%
  inner_join(adults.vs.children.m, by = c("ID","nameofG"), suffix = c(".Adults.vs.Children.F", ".Adults.vs.Children.M")) %>%
  inner_join(elderly.vs.children.f, by = c("ID","nameofG")) %>%
  inner_join(elderly.vs.children.m, by = c("ID","nameofG"), suffix = c(".Elderly.vs.Children.F", ".Elderly.vs.Children.M")) %>%
  inner_join(elderly.vs.adults.f, by = c("ID","nameofG")) %>%
  inner_join(elderly.vs.adults.m, by = c("ID","nameofG"), suffix = c(".Elderly.vs.Adults.F", ".Elderly.vs.Adults.M"))
  #full_join(m.vs.f.children, by = c("ID","nameofG")) %>%
  #full_join(m.vs.f.adults, by = c("ID","nameofG"), suffix = c(".M.vs.F.Children", ".M.vs.F.Adults")) %>%
  #full_join(dplyr::rename(m.vs.f.elderly, logFC.M.vs.F.Elderly = logFC, logCPM.M.vs.F.Elderly = logCPM, F.M.vs.F.Elderly = F, PValue.M.vs.F.Elderly = PValue, FDR.M.vs.F.Elderly = FDR), by = c("ID","nameofG"))

out <- read.delim(here("data", "ruv_counts_curettage.tsv"))

rownames(out) <- out$ID
out <- out[,-c(1:2)]
out <- as.data.frame(t(out))

out$label <- rownames(out)

out3 <- phenodata %>%
   full_join(out, by = "label") %>%
   pivot_longer(-colnames(phenodata), names_to = "genes", values_to = "values") %>%
   group_by(Group, sex, genes) %>%
   summarise(mean=mean(values)) %>%
   pivot_wider(id_cols = "genes", names_from = c("Group","sex"), values_from = "mean") %>%
   inner_join(out2, by = c("genes" = "ID")) %>%
   select(-starts_with(c("logCPM.", "LR."))) %>%
   select(ID = genes, nameofG,RUVseq_Adults_M = Adults_M, RUVseq_Adults_F = Adults_F, 
          RUVseq_Children_F = Children_F, RUVseq_Children_M = Children_M, RUVseq_Elderly_M = Elderly_M,
          RUVseq_Elderly_F = Elderly_F, everything())

write.table(out3, here("results",
                       "RUVSeq_k4_diff_expression_byAge_bySex_curettage.tsv"),
            sep = "\t", quote = F, row.names = F)
```

```{r Join RUVseq results only by Age groups}
children.vs.adults <- read.delim(here("results", "diff_express_edgeR",
"DEGs_Children.vs.Adults_glmLRT_RUVg_k4_empirical_control_all_groups_102samples.tsv"))

children.vs.adults <- children.vs.adults %>% 
  mutate(ctrl = "Children", treat = "Adults")

children.vs.elderly <- read.delim(here("results", "diff_express_edgeR", "DEGs_Children.vs.Elderly_glmLRT_RUVg_k4_empirical_control_all_groups_102samples.tsv"))

children.vs.elderly <- children.vs.elderly %>% 
  mutate(ctrl = "Children", treat = "Elderly")

adults.vs.elderly <- read.delim(here("results", "diff_express_edgeR", "DEGs_Adults.vs.Elderly_glmLRT_RUVg_k4_empirical_control_all_groups_102samples.tsv"))

adults.vs.elderly  <- adults.vs.elderly  %>% 
  mutate(ctrl = "Adults", treat = "Elderly")

out <- rbind(children.vs.adults, children.vs.elderly, adults.vs.elderly)
out <- out %>%
  select(ID, nameofG, ctrl, treat, everything()) %>% 
  select(-c(6:7)) %>% 
  arrange(ID)
  
write.table(out, here("results",
                       "RUVSeq_k4_diff_expression_byAge_curettage_tidy.tsv"),
            sep = "\t", quote = F, row.names = F)
```



```{r}
rownames(tmm) <- tmm$ID

#filter low expression genes
keep <- rowSums(tmm > 1) > ncol(tmm[,-c(1:2)])*0.5
subtmm <- tmm[keep,]

hgnc <- subtmm[,1:2]

subtmm.t <- as.data.frame(t(subtmm[,-c(1:2)]))

subtmm.t$label <- rownames(subtmm.t)
subtmm.t <- merge(phenodata_curettage, subtmm.t, by="label")

subtmm.t <- subtmm.t[,c(colnames(subtmm.t[,1:7]),selectedGenes)]
#pca
prcom.subtmm <- prcomp(subtmm.t[,-c(1:7)])

autoplot(prcom.subtmm , subtmm.t, shape = 'sex', 
         colour= 'Group', size=3) +
  scale_color_manual(values= brewer.pal(3,"Set2")) +
  labs(title = "TMM") +
  theme_bw(base_size = 15,)

prcom.subtmm <- prcomp(subtmm.t[selectedGenes,-c(1:7)])

autoplot(prcom.subtmm , subtmm.t, shape = 'sex', 
         colour= 'Group', size=3) +
  scale_color_manual(values= brewer.pal(3,"Set2")) +
  labs(title = "TMM") +
  theme_bw(base_size = 15,)



settmm <- newSeqExpressionSet(as.matrix(subtmm[,-c(1:2)]),
                           phenoData = data.frame(phenodata_curettage$Group,
                                                  row.names=colnames(subtmm[,-c(1:2)])))
colors <- brewer.pal(3, "Set2")

png(width = 888, height = 500, "RLE_TMM.png")
plotRLE(settmm, col = colors[factor(phenodata_curettage$Group, 
                                    levels= c("Children","Adults", "Elderly"))], 
        outline=FALSE, ylim=c(-4, 4), main = "TMM ",
        ylab= "Relative Log Expression")
legend("topright",legend = c("Children", "Adults", "Elderly"), 
       col = colors, pch = 15,pt.cex = 1.5, cex = 1.2, ncol=1)
dev.off()

plotPCA(settmm, col=colors[as.factor(phenodata_curettage$Group)], cex=1.2)

selectedGenes <- names(sort(apply(counts(settmm), 1, var), decreasing = T)[1:2000])
pheatmap(counts(settmm[selectedGenes,]), 
             scale = 'row',
             #cutree_cols = 3, 
             show_rownames = FALSE,)


```


```{r TMM gene expression 96 samples}
# TMM counts with y DGEList object from standard diff express analysis
tmm <- as.data.frame(cpm(y))

tmm <- as.data.frame(t(tmm))

tmm$label <- rownames(tmm)
tmm <- merge(phenodata, tmm, by="label")
x <- tmm %>% 
  pivot_longer(-colnames(tmm[,1:7]), 
               names_to = "ensemblID", values_to = "values") %>%
  left_join(hgnc, by= c("ensemblID" = "ID"))

genes <- c("ACE2","TMPRSS2","FURIN","ANPEP","CTSB","CTSL")

x %>%
  filter(nameofG %in% genes) %>%
  mutate(Group = fct_relevel(Group, "Children","Adults","Older")) %>%
  mutate(nameofG = fct_relevel(nameofG, genes)) %>%
  ggplot(aes(x = Group, y = values, fill = sex), cex=1.5) +
  geom_boxplot(position= "dodge", outlier.shape=NA) +
  facet_wrap(vars(nameofG),nrow=2) +
  scale_y_continuous(trans='log2') +
  ylab("log2 TMM counts")+
  labs(caption = "Gene expression of six genes involved with Sars-CoV-2 cell entry. 
       Gene expression was calculated based on 96 samples after
       removing outliers detected with MDP analysis. Outliers were 
       defineid with 1.5*IQR of samples score from MDP") +
  theme_bw(base_size = 12)
```

## IFN and ISG Children vs Elderly M and F
```{r}
deg <- read.delim(here("results",
                       "RUVSeq_k4_diff_expression_byAge_bySex_curettage.tsv"))
isg <- read.delim(here("data", "isg_genes.txt"))

deg.tidy <- deg %>% 
  select(-starts_with("RUV")) %>% 
  pivot_longer(-c(1:2),names_to = "method", values_to = "result") %>% 
  separate(method, c("metric","treat","vs","ctrl","sex"), sep = "([.])") %>% 
  select(-vs) %>% 
  pivot_wider(names_from = metric, values_from = result) %>% 
  select(ID, nameofG, ctrl, everything())

write.table(deg.tidy, here("results", 
                    "RUVSeq_k4_diff_expression_byAge_bySex_curettage_tidy.tsv"),
            sep = "\t", quote = F, row.names = F)

norm.count <- deg %>% 
  select(ID, nameofG,starts_with("RUV"))

write.table(norm.count, here("data", 
                    "ruv_counts_curettage_byAge_bySex.tsv"),
            sep = "\t", quote = F, row.names = F)

isg.deg <- deg.tidy %>% 
  filter(nameofG %in% isg$genes) %>% 
  filter(ctrl == "Children", treat == "Elderly") %>% 
  mutate(deg = case_when(logFC > 1 & FDR < 0.05 ~ "up",
                         logFC < -1 & FDR < 0.05 ~ "down"),) 

write.table(isg.deg, here("results", "degs_ISGs.tsv"), 
            sep = "\t", quote = F, row.names = F)

isg.deg %>% 
  filter(deg %in% c("up","down")) %>% 
  group_by(sex, deg) %>% 
  summarise(n = n()) %>% 
  ggplot(aes(x = deg, y = n)) +
  geom_bar(aes(fill = sex),stat = "identity", position = "dodge",
           color = "black", width = 0.6) +
  scale_fill_manual(values =c("#ef8a62","#67a9cf")) +
  scale_x_discrete (labels = c("Down", "Up")) + 
  labs(title = "ISGs    |logFC| > 1 & FDR < 0.05",y = "# DEGs", x = "") +
  theme_bw() +
  theme(text = element_text(size = 13))

```

## IFN and ISG Children vs Elderly
```{r}
deg.tidy <- read.delim(here("results",
                       "RUVSeq_k4_diff_expression_byAge_curettage_tidy.tsv"))
isg <- read.delim(here("data", "isg_genes.txt"))

isg.deg <- deg.tidy %>% 
  filter(nameofG %in% isg$genes) %>% 
  filter(ctrl == "Children", treat == "Elderly") %>% 
  mutate(deg = case_when(logFC > 1 & FDR < 0.05 ~ "up",
                         logFC < -1 & FDR < 0.05 ~ "down"),) 

write.table(isg.deg, here("results", "degs_ISGs_byAgeGroup.tsv"), 
            sep = "\t", quote = F, row.names = F)

isg.deg %>% 
  filter(deg %in% c("up","down")) %>% 
  group_by(deg) %>% 
  summarise(n = n()) %>% 
  ggplot(aes(x = deg, y = n)) +
  geom_bar(aes(fill = deg),stat = "identity", color = "black", width = 0.5) +
  scale_fill_manual(values =c("#ef8a62","#67a9cf")) +
  scale_x_discrete (labels = c("Down", "Up")) + 
  labs(title = "ISGs - |logFC| > 1 & FDR < 0.05\nChildren vs Elderly",
       y = "# DEGs", x = "") +
  theme_bw() +
  theme(legend.position="none" ,text = element_text(size = 13))

```
